### PR TITLE
feat: display other currencies for non evm accounts.

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

This PR enables the user's selected currency to be displayed for non evm accounts. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28963?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/611

## **Manual testing steps**

1. Create a BTC account with mainnet funds on flask.
2. Go to settings then general and select a currency
3. See that the currency is displayed in the overview and account list menu

### **Before**

Screenshot 2024-12-12 at 20 37 44
Screenshot 2024-12-12 at 20 37 36
Screenshot 2024-12-12 at 20 37 33

### **After**

Screenshot 2024-12-12 at 20 36 27
Screenshot 2024-12-12 at 20 36 23

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/selectors/multichain.ts [hunk 1]
```diff
@@ -256,20 +256,10 @@ export function getMultichainNativeCurrency(
 
 export function getMultichainCurrentCurrency(
   state: MultichainState,
-  account?: InternalAccount,
+  _account?: InternalAccount,
```
Comment (gantunesr) [inline ui/selectors/multichain.ts:259]: Should this be removed since it is not being used anymore?

File: ui/selectors/multichain.ts [hunk 2]
```diff
@@ -256,20 +256,10 @@ export function getMultichainNativeCurrency(
...
-    return currentCurrency;
-  }
-
-  // For non-EVM:
-  // To mimic `getCurrentCurrency` we only consider fiat values, otherwise we
-  // fallback to the current ticker symbol value
-  return currentCurrency && currentCurrency.toLowerCase() === 'usd'
-    ? 'usd'
-    : getMultichainProviderConfig(state, account).ticker;
+  return currentCurrency;
```
Comment (gantunesr) [inline ui/selectors/multichain.ts:262]: nit

```ts
  return getCurrentCurrency(state);
```

File: ui/selectors/multichain.ts [hunk 3]
```diff
@@ -387,7 +380,7 @@ function getNonEvmCachedBalance(state: MultichainState) {
       network.chainId as MultichainNetworks
     ]?.[0];
 
-  return balances?.[account.id]?.[asset]?.amount;
+  return balances?.[selectedAccount.id]?.[asset]?.amount ?? 0;
```
Comment (gantunesr) [inline ui/selectors/multichain.ts:383]: I wonder if we can handle this in a better than just a conditional chaining. Maybe adding some guards that will console.error if an attr is falsy?

**Comment by gantunesr** [comment]
Can you add screenshots and steps?

File: ui/selectors/multichain.ts [hunk 4]
```diff
@@ -387,7 +376,21 @@ function getNonEvmCachedBalance(state: MultichainState) {
...
     ]?.[0];
 
-  return balances?.[account.id]?.[asset]?.amount;
+  if (!asset) {
+    console.warn('Could not find asset type for network:', network);
+  }
+
+  const balancesForAccount = balances?.[selectedAccount.id];
+  if (!balancesForAccount) {
+    console.warn('Could not find balances for account:', selectedAccount);
```
Comment (montelaidev) [inline ui/selectors/multichain.ts:385]: Changed to console.warn to avoid causing the e2e tests to fail.

**Review by gantunesr — APPROVED** [review]

**Review by vinnyhoward — APPROVED** [review]

**Pull request merged**